sparkly/utils.py

```diff
@@ -59,6 +64,89 @@ def absolute_path(file_path, *rel_path):
     )
 
 
+def chunks(seq, n):
+    """Iterate over seq by chunks of n elements.
+
+    >>> list(chunks([1, 2, 3, 4, 5], 2))
+    [[1, 2], [3, 4], [5]]
+    >>> list(chunks([], 2))
+    []
+    >>> list(chunks([1], 2))
+    [[1]]
+    """
+    it = iter(seq)
+    while True:
+        chunk = list(islice(it, n))
+        if not chunk:
+            break
+        yield chunk
+
+
+def type_has(t, required_fields):
```

Comment:
Other than the fact that `dataframe_has` seems to be a better name it doesn't save us anything else (just the `.schema` part), so I'm not sure it's worth to have to maintain it (and write tests etc). Maybe rename `type_has` to `datatype_has`?